### PR TITLE
Add multichain hybrid guide

### LINKED ISSUE #47: MultiChain on-prem hybrid
multichain hybrid installation

### PR DESCRIPTION
**Pull request by akegaviar**
Add the MultiChain hybrid deployment guide under "Guides".
Crosslink with glossary.
Closes #47

### CODE REVIEW

File: docs/guides/multichain-hybrid.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+where
+
+* CHAIN_NAME — your MultiChain cloud deployment chain name. Available under **Credentials** —> **Chain name**.
+* HOSTNAME — your MultiChain cloud deployment hostname. Available under **Credentials** —> **Host**.
+* PORT — your MultiChain cloud deployment port. Available under **Credentials** —> **Port**.
+
+Command example:
+
+```
+multichaind nw-784-155-2@nd-339-567-264.rg-441-738.p2pify.com:7447 -daemon
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: 1) I suggest to use dummy data here. E.g. everywhere else I use 'nw-123-456-7', 'nd-123-456-789' and so on.
2) The node address will be nd-XXX-XXX-XXX.p2pify.com, e.g.:
`$ ./multichaind nw-123-456-7@nd-123-456-789.p2pify.com:7447`

File: docs/guides/multichain-hybrid.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+Run:
+
+```
+multichaind CHAIN_NAME@HOSTNAME:PORT -daemon
+```
+
+where
+
+* CHAIN_NAME — your MultiChain cloud deployment chain name. Available under **Credentials** —> **Chain name**.
+* HOSTNAME — your MultiChain cloud deployment hostname. Available under **Credentials** —> **Host**.
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: There is no such field anymore. The hostname should be extracted from 'RPC endpoint' field.

File: docs/guides/multichain-hybrid.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+* PORT — your MultiChain cloud deployment port. Available under **Credentials** —> **Port**.
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: There is no such field anymore. We are going to introduce a proper way of hybrid deployment soon. This guide is an intermediate step, so we just say the port is 7447.

File: docs/guides/multichain-hybrid.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+* Your on-premises node's wallet address.
+
+Output example:
+
+```
+MultiChain 2.0.2 Daemon (Community Edition, latest protocol 20010)
+
+Starting up node...
+
+Retrieving blockchain parameters from the seed node nd-339-567-264.rg-441-738.p2pify.com:7447 ...
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: I suggest to use dummy data here

File: docs/guides/multichain-hybrid.md [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+Blockchain successfully initialized.
+
+Please ask blockchain admin or user having activate permission to let you connect and/or transact:
+multichain-cli nw-784-155-2 grant 14SW7CidNbktZxkTSzi52iLvXviHyPebqCaW1q connect
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: I suggest to use dummy data here

File: docs/guides/multichain-hybrid.md [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+multichain-cli nw-784-155-2 grant 14SW7CidNbktZxkTSzi52iLvXviHyPebqCaW1q connect,send,receive
+```
+
+### 6. Grant permissions to your on-premises MultiChain node
+
+You can use tools like [curl](https://curl.haxx.se/) or [Postman](https://www.getpostman.com/) to invoke [MultiChain API methods](https://www.multichain.com/developers/json-rpc-api/).
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: Perhaps we can link to our guide how to connect to MultiChain node.

File: docs/guides/multichain-hybrid.md [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+
+Sending a curl request from terminal:
+
+```
+curl HOSTNAME -u "USERNAME:PASSWORD" -d {"method":"grant","params":["WALLET_ADDRESS","connect,send,receive"],"chain_name":"CHAIN_NAME"}'
+```
+
+where
+
+* HOSTNAME — your MultiChain cloud deployment hostname. Available under **Credentials** —> **Host**.
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: See above regarding the absence of this field in the upcoming release.

File: docs/guides/multichain-hybrid.md [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+* USERNAME — your MultiChain cloud deployment username. Available under **Credentials** —> **User**.
+* PASSWORD — your MultiChain cloud deployment password. Available under **Credentials** —> **Password**.
+* WALLET_ADDRESS — your MultiChain on-premises node's wallet address. You received the wallet address at the end of [Step 5](multichain-hybrid#_5-initialize-your-on-premises-multichain-node).
+* CHAIN_NAME — your MultiChain cloud deployment chain name. Available under **Credentials** —> **Chain name**.
+
+Command example:
+
+```
+curl https://nd-339-567-264.p2pify.com -u "modest_cori:ought vilify parcel urging dime sixth" -d '{"method":"grant","params":["14SW7CidNbktZxkTSzi52iLvXviHyPebqCaW1q","connect,send,receive"],"chain_name":"nw-784-155-2"}'
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: I suggest to use dummy data here

File: docs/guides/multichain-hybrid.md [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+
+Sending a curl request from terminal:
+
+```console
+curl HOSTNAME -u "USERNAME:PASSWORD" -d '{"method":"addnode","params":["ON_PREM_IP:PORT","add"],"chain_name":"CHAIN_NAME"}'
+```
+
+where
+
+* HOSTNAME — your MultiChain cloud deployment hostname. Available under **Credentials** —> **Host**.
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: Same as above

File: docs/guides/multichain-hybrid.md [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+* USERNAME — your MultiChain cloud deployment username. Available under **Credentials** —> **User**.
+* PASSWORD — your MultiChain cloud deployment password. Available under **Credentials** —> **Password**.
+* ON_PREM_IP — your MultiChain on-premises machine's IP address.
+* PORT — your MultiChain on-premises machine's port.
+* CHAIN_NAME — your MultiChain cloud deployment chain name. Available under **Credentials** —> **Chain name**.
+
+Command example:
+
+```console
+curl https://nd-339-567-264.p2pify.com -u "modest_cori:ought vilify parcel urging dime sixth" -d '{"method":"addnode","params":["178.62.100.80:7447","add"],"chain_name":"nw-784-155-2"}'
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: I suggest to use the dummy data here

File: docs/guides/multichain-hybrid.md [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+Run:
+
+```
+multichaind CHAIN_NAME@HOSTNAME:PORT -daemon
+```
+
+where
+
+* CHAIN_NAME — your MultiChain cloud deployment chain name. Available under **Credentials** —> **Chain name**.
+* HOSTNAME — your MultiChain cloud deployment hostname. Available under **Credentials** —> **Host**.
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: Same as above

File: docs/guides/multichain-hybrid.md [hunk 12]
```diff
@@ -0,0 +1,264 @@
...
+* PORT — your MultiChain cloud deployment port. Available under **Credentials** —> **Port**.
+
+Command example:
+
+```console
+multichaind nw-784-155-2@nd-339-567-264.rg-441-738.p2pify.com:7447 -daemon
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: I suggest to use dummy data here

File: docs/guides/multichain-hybrid.md [hunk 13]
```diff
@@ -0,0 +1,264 @@
...
+where
+
+* CHAIN_NAME — your MultiChain cloud deployment chain name. Available under **Credentials** —> **Chain name**.
+* IP_ADDRESS — your on-premises machine name's IP address.
+* PORT — your on-premises node's peer port.
+
+Example command:
+
+```
+multichain-cli nw-784-155-2@178.62.100.80:7447
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: I suggest to use dummy data here

File: docs/guides/multichain-hybrid.md [hunk 14]
```diff
@@ -0,0 +1,264 @@
...
+
+```console
+multichain-cli nw-784-155-2@nd-339-567-264.rg-441-738.p2pify.com:7447
+```
+
+### Run commands in interactive mode
+
+Once in interactive mode, run any [MultiChain JSON-RPC command](https://www.multichain.com/developers/json-rpc-api/).
+
+Examples:
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: Perhaps we should:
1) Include an interactive interpreter prefix
2) Example outputs (trimmed where required)

File: docs/guides/multichain-hybrid.md [hunk 15]
```diff
@@ -0,0 +1,264 @@
...
+1. With Chainstack, create a [Consortium](/projects/consortium) project.
+2. With Chainstack, deploy a MultiChain network in cloud.
+3. With Chainstack, get your cloud MultiChain node access information.
+4. On-premises, install MultiChain.
+5. On-premises, initialize your MultiChain node.
+6. From your cloud MultiChain node, grant permissions to your on-premises MultiChain node's wallet address.
+7. From your cloud MultiChain node, add your on-premises MultiChain node to the network.
+8. From your on-premises MultiChain node, connect to the MultiChain network.
+
+## Step-by-step
```
Comment (easeev) [inline docs/guides/multichain-hybrid.md]: Maybe this is redundant and we can have exact steps on h2 instead

File: docs/guides/multichain-hybrid.md [hunk 16]
```diff
@@ -0,0 +1,264 @@
...
+
+Run:
+
+```
+multichaind CHAIN_NAME@HOSTNAME:PORT -daemon
+```
+
+where
+
+* CHAIN_NAME — your MultiChain cloud deployment chain name. Available under **Credentials** —> **Chain name**.
```
[the remark this answers is not in the document]
Comment (akegaviar) [reply docs/guides/multichain-hybrid.md]: Good catch actually. We must use `>` which is the tech writing standard. Will fix.

File: docs/guides/multichain-hybrid.md (hunk 1 above)
Comment (akegaviar) [reply docs/guides/multichain-hybrid.md]: it's dummy data in all of the examples in my commit but I agree that making it obvious with 123456 is better. Will fix in all examples

File: docs/guides/multichain-hybrid.md (hunk 14 above)
Comment (akegaviar) [reply docs/guides/multichain-hybrid.md]: 1. Done
2. I played around with the output trimmed and untrimmed and went with the untrimmed version. Let me know if you'd like a trimmed one and what the cutoff line should be as I'm not sure myself

File: docs/guides/multichain-hybrid.md (hunk 15 above)
Comment (akegaviar) [reply docs/guides/multichain-hybrid.md]: One is a part of an overview and the other starts the detailed process. I don't think this is redundant but we can discuss.

**Pull request merged**